src/EnergyPlus/Boilers.cc

```diff
@@ -210,18 +210,14 @@ void GetBoilerInput(EnergyPlusData &state)
         thisBoiler.Type = DataPlant::PlantEquipmentType::Boiler_Simple;
 
         // Validate fuel type input
-        bool FuelTypeError(false);
-        UtilityRoutines::ValidateFuelTypeWithAssignResourceTypeNum(
-            state.dataIPShortCut->cAlphaArgs(2), thisBoiler.BoilerFuelTypeForOutputVariable, thisBoiler.FuelType, FuelTypeError);
-        if (FuelTypeError) {
+        thisBoiler.FuelType = static_cast<DataGlobalConstants::ResourceType>(
+            getEnumerationValue(DataGlobalConstants::ResourceTypeNamesUC, UtilityRoutines::MakeUPPERCase(state.dataIPShortCut->cAlphaArgs(2))));
```

Comment:
Based on the boiler (haven't looked at other components) the code here would allow anything in the resource type list to be a valid fuel to power the boiler. The only place the limits the choices here is the key choice list in the IDD/schema for this input field. We've relied on that in other places to weed out nonsense inputs but usually the key choice list and the enum are the same list. Here we're relying on the IDD/schema to restrict to a subset. I suppose that's ok. But if you're just reading the source code here, you don't know that the full list of `ResourceType` is not valid for this object.

Reply:
🤔 Maybe we do want to consider this in this PR after all.